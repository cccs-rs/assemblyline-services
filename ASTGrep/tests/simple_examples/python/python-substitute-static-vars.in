var1 = 1
var2 = "aaaa"
var3 = var1

funcx(var2)
funcx(var2, var2)
funcx(var2 + var2)
[var1]
[var3]

func = 123

# Functions not supported yet
def func(var1):
    r = var1
    return r

def f2(var1):
    x = var1 + var2
    return var1

var20 = 2

# TODO: this should be ignored
x = 7
for i in range(3):
    x = i
print(x)

global var2
var1 += "a"

aaaa = 7
def f(b, c, aaaa):
    x=aaaa

def f(b, c, aaaa=7):
    x=aaaa

def f(b, c, d=aaaa):
    x=aaaa
