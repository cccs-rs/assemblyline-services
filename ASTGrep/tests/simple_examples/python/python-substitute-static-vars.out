var1 = 1
var2 = "aaaa"
var3 = 1

funcx("aaaa")
funcx("aaaa", "aaaa")
funcx('aaaaaaaa')
[1]
[1]

func = 123


def func(var1):
    r = var1
    return r

def f2(var1):
    x = var1 + "aaaa"
    return var1

var2 = 2


x = 7
for i in range(3):
    x = i
print(7)
